### PR TITLE
Added malletmodel2ldamodel transformation function to mallet wrapper

### PR DESCRIPTION
**Pull request by dsquareindia**
This is concerning [this](https://groups.google.com/d/msg/gensim/rVHkzkFRNdI/p9uTDJ3ABAAJ) thread in the mailing list. @piskvorky @tmylk could you please check?

### CODE REVIEW

File: gensim/models/wrappers/ldamallet.py [hunk 1]
```diff
@@ -346,3 +347,22 @@ def read_doctopics(self, fname, eps=1e-6, renorm=True):
...
+
+    Args:
+    ----
+    mallet_model : Trained mallet model
+
+    Returns:
+    -------
+    model_gensim : LdaModel instance; copied gensim LdaModel
+    """
+    model_gensim = LdaModel(id2word=mallet_model.id2word, num_topics=mallet_model.num_topics, alpha=mallet_model.alpha, iterations=100)
```
Comment (bhargavvader) [inline gensim/models/wrappers/ldamallet.py]: Just curious, is there no way to pick up the number of iterations from the mallet model? I notice you've kept it as the default `100` right now.

Comment (dsquareindia) [inline gensim/models/wrappers/ldamallet.py]: Yes even I thought of doing that but I instead just followed what @piskvorky wrote in the mailing list. Both are giving similar results though. I guess I'll switch this to `mallet_model.iterations` for better resemblance to the mallet model.

File: gensim/test/test_ldamallet_wrapper.py [hunk 2]
```diff
@@ -95,6 +95,18 @@ def testSparseTransform(self):
...
 
+    def testMallet2Model(self):
+        if not self.mallet_path:
+            return
+        passed = False
+        tm1 = ldamallet.LdaMallet(self.mallet_path, corpus=corpus, num_topics=2, id2word=dictionary)
+        tm2 = ldamallet.malletmodel2ldamodel(tm1)
+        for document in corpus:
+            self.assertAlmostEqual(tm1[document][0][1], tm2[document][0][1], places=1)
+            self.assertAlmostEqual(tm1[document][1][1], tm2[document][1][1])
```
Comment (dsquareindia) [inline gensim/test/test_ldamallet_wrapper.py]: @piskvorky I'm not too sure about this test. Values sometimes are differing at the first place too.

File: gensim/models/wrappers/ldamallet.py [hunk 3]
```diff
@@ -346,3 +347,23 @@ def read_doctopics(self, fname, eps=1e-6, renorm=True):
...
+    Args:
+    ----
+    mallet_model : Trained mallet model
+
+    Returns:
+    -------
+    model_gensim : LdaModel instance; copied gensim LdaModel
+    """
+    model_gensim = LdaModel(id2word=mallet_model.id2word, num_topics=mallet_model.num_topics,
+                            alpha=mallet_model.alpha, iterations=mallet_model.iterations)
```
Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: `iterations` means something very different between Variational Bayes (gensim) and Gibbs sampling (Mallet). There is no 1:1 correspondence, the semantics are different.

File: gensim/test/test_ldamallet_wrapper.py (hunk 2 above)
Comment (piskvorky) [inline gensim/test/test_ldamallet_wrapper.py]: Try more accurate `gamma_threshold` and `iterations`. But I don't think the results can really be made identical -- the inference algorithms are completely different.

I think the best we can hope for is a sanity check -- topics in same order, on some (model, query_document) pair where it's clear what the topic order should be, and where any deviation in topic order is clearly an error.

File: gensim/models/wrappers/ldamallet.py (hunk 3 above)
Comment (dsquareindia) [inline gensim/models/wrappers/ldamallet.py]: Should I let it remain 100 then? Equating both is not correct then....

Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: Yes, probably leave the default. Or give an option (fnc args) to the user to set `gamma_threshold`/`iterations` explicitly.

File: gensim/models/wrappers/ldamallet.py [hunk 4]
```diff
@@ -346,3 +347,23 @@ def read_doctopics(self, fname, eps=1e-6, renorm=True):
...
+
+    Args:
+    ----
+    mallet_model : Trained mallet model
+
+    Returns:
+    -------
+    model_gensim : LdaModel instance; copied gensim LdaModel
+    """
+    model_gensim = LdaModel(id2word=mallet_model.id2word, num_topics=mallet_model.num_topics,
```
Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: We don't use this type of indent in gensim -- it's hard to maintain when editing & also visually ugly.

Please use normal "hanging indent" with 4 spaces: https://www.python.org/dev/peps/pep-0008/#indentation

**Comment by dsquareindia** [comment]
@piskvorky I've addressed the comments. Could you please check?

File: gensim/models/wrappers/ldamallet.py [hunk 5]
```diff
@@ -358,12 +358,16 @@ def malletmodel2ldamodel(mallet_model):
...
+    iterations : number of iterations to be used for LdaModel training.
 
     Returns:
     -------
     model_gensim : LdaModel instance; copied gensim LdaModel
     """
-    model_gensim = LdaModel(id2word=mallet_model.id2word, num_topics=mallet_model.num_topics,
-                            alpha=mallet_model.alpha, iterations=mallet_model.iterations)
+    model_gensim = LdaModel(
+                id2word=mallet_model.id2word, num_topics=mallet_model.num_topics,
```
Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: PEP8: Hanging indent of 4 spaces.

File: gensim/models/wrappers/ldamallet.py [hunk 6]
```diff
@@ -358,12 +358,16 @@ def malletmodel2ldamodel(mallet_model):
     Args:
     ----
     mallet_model : Trained mallet model
+    gamma_threshold : To be used for LdaModel training.
```
Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: Misleading comment: there is no "training", the model is transferred from Mallet. These parameters only affect inference, model is unchanged.

File: gensim/models/wrappers/ldamallet.py [hunk 7]
```diff
@@ -358,12 +358,16 @@ def malletmodel2ldamodel(mallet_model):
...
+    iterations : number of iterations to be used for LdaModel training.
```
Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: dtto

**Comment by dsquareindia** [comment]
@piskvorky I've made the changes. Could you please check?

**Pull request merged**